### ISSUE TITLE
set password problem -  DSID-031A126C, problem 5003

### ISSUE THREAD

**Issue by mowglis** [cannatag/ldap3#1157]
Hi,
after succesfly ceated user:

attributes={
                 'objectClass': ['top', 'person', 'organizationalPerson', 'user'],
                 'cn': f'{first_name} {last_name}',
                 'givenName': first_name,
                 'sn': last_name,
                 'sAMAccountName': username,
                 'userPrincipalName': f'{username}@{self.ad_domain}',
                 'displayName': f'{first_name} {last_name}',
                  'homeDirectory': home_directory,
                 'homeDrive': home_drive,
                 'profilePath': profile_path,
                 'mail': user_email
             }
conn.add(user_dn, attributes=attributes)

I try set password and unlock account with this commands:
 # Update password
 conn.extend.microsoft.unlock_account(user=user_dn)
 conn.extend.microsoft.modify_password(user_dn, password)
 conn.modify(user_dn, changes={"userAccountControl": (MODIFY_REPLACE, [512])})

I recive this error:
{
    'result': 53,
    'description': 'unwillingToPerform',
    'dn': '',
    'message': '0000052D: SvcErr: DSID-031A126C, problem 5003 (WILL_NOT_PERFORM), data 0\n\x00',
    'referrals': None,
    'type': 'modifyResponse'
}
AD server is Windows 2019.... Please, can You help me, where is problem to set password? Thanks.

**Comment by cannatag** [comment]
Hi. Are using a secure connection to the server? It won’t accept a modify password operation on a cleartext connection.Bye,GiovanniIl giorno 28 ago 2024, alle ore 22:30, Ondřej Rusek ***@***.***> ha scritto:﻿
Hi,
after succesfly ceated user:
attributes={
'objectClass': ['top', 'person', 'organizationalPerson', 'user'],
'cn': f'{first_name} {last_name}',
'givenName': first_name,
'sn': last_name,
'sAMAccountName': username,
'userPrincipalName': f'{username}@{self.ad_domain}',
'displayName': f'{first_name} {last_name}',
'homeDirectory': home_directory,
'homeDrive': home_drive,
'profilePath': profile_path,
'mail': user_email
}
conn.add(user_dn, attributes=attributes)
I try set password and unlock account with this commands:
Update password
conn.extend.microsoft.unlock_account(user=user_dn)
conn.extend.microsoft.modify_password(user_dn, password)
conn.modify(user_dn, changes={"userAccountControl": (MODIFY_REPLACE, [512])})
I recive this error:
{
'result': 53,
'description': 'unwillingToPerform',
'dn': '',
'message': '0000052D: SvcErr: DSID-031A126C, problem 5003 (WILL_NOT_PERFORM), data 0\n\x00',
'referrals': None,
'type': 'modifyResponse'
}
AD server is Windows 2019.... Please, can You help me, where is problem to set password? Thanks.

**Comment by mowglis** [comment]
> Hi. Are using a secure connection to the server? It won’t accept a modify password operation on a cleartext connection.

Yes, good question..... not. My connection is create with this commands:

**ad_server = 'dc01.ad.gybon.cz'
server = Server(ad_server, get_info=ALL, use_ssl=False)**

This is not secure, I think.... But with use_ssl=True, I got error:
**ldap3.core.exceptions.LDAPSocketSendError: socket sending error[Errno 104] Connection reset by peer**

**Comment by cannatag** [comment]
Secure port should be 636, but it could be different for your server. You can also try to upgrade the cleartext connection to a secure one with the StartTLS operation. Try with conn.startTLS() before changing the password.Bye,GiovanniIl giorno 28 ago 2024, alle ore 22:45, Ondřej Rusek ***@***.***> ha scritto:﻿

Hi. Are using a secure connection to the server? It won’t accept a modify password operation on a cleartext connection.

Yes, good question..... not. My connection is create with this commands:
ad_server = 'dc01.ad.gybon.cz'
server = Server(ad_server, get_info=ALL, use_ssl=False)
This is not secure, I think.... But with use_ssl=True, I got error:
ldap3.core.exceptions.LDAPSocketSendError: socket sending error[Errno 104] Connection reset by peer

**Comment by mowglis** [comment]
Thanks for your help, my problem will be in SSL communication with AD server.
